tests/test_offload.py

```diff
@@ -49,6 +53,22 @@ def test_offload_state_dict(self):
                 self.assertTrue(os.path.isfile(weight_file))
                 # TODO: add tests on the fact weights are properly loaded
 
+    def test_offload_weight(self):
+        dtypes = [torch.float16, torch.float32]
+        if is_torch_version(">=", "1.10"):
+            dtypes.append(torch.bfloat16)
+
+        for dtype in dtypes:
+            weight = torch.tensor([[1, 2, 3], [4, 5, 6]], dtype=dtype)
```

Comment:
Since we care about the prevision here, would it be a good idea to test these with floats that marginally get close to the float16 reach to ensure they're lossless? 

E.g.:
```python
weight = torch.rand((2,3), dtype=dtype)
```

Reply:
Probably yes 😅